rules/common_settings.bzl

```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common build setting rules that return a BuildSettingInfo with the value of the build setting.
+For label-typed settings, use the native label_flag and label_setting rules."""
+
+BuildSettingInfo = provider(fields = ["value"])
+
+def _impl(ctx):
+    return BuildSettingInfo(value = ctx.build_setting_value)
+
+int_flag = rule(
+    implementation = _impl,
```

Comment:
Could you add doc strings to these rules, and then generate documentation for them using stardoc? (The doc generation and check-in could be a future pull request, but I'd like to see the docstrings here at least :)

Reply:
PTAL a look at docstrings and let me know if you think they need more! They're all pretty simple but I've been stewing on these concepts for ages so would understand if they don't come across as simple to all.